content/news/2017-02-20-mapstruct-1_2_0_Beta1-is-out-with-lombok-support-and-direct-field-access.md

```diff
@@ -0,0 +1,195 @@
+}
+{{< /prettify >}}
+
+If the "element type" differs between source and target stream/collection (`Order` vs. `OrderDto`),
+`Stream#map()` will be used to call the right element mapping method (`orderToOrderDto()`) in the generated code.
+
+### Mappings using direct field access
+
+Traditionally MapStruct relied on the JavaBeans convention and called getter and setter methods for propagating property values from source to target.
+Sometimes writing all the getters and setters may be a bit cumbersome, though, e.g. when it comes to "dumb" DTOs which should just hold a couple of property values.
+
+Using public fields is perfectly fine in such case, which is why direct field access is [supported by MapStruct now](https://github.com/mapstruct/mapstruct/issues/557).
+I.e. if your source or target beans do not define getters or setters for the properties but declare public fields,
+the generated code will directly access the fields.
+Note that no reflection will be used, i.e. we won't make private fields accessible.
+
+You can find a small example demonstrating field-based mapping [here](https://github.com/mapstruct/mapstruct-examples/tree/master/mapstruct-field-mapping/).
+
+### Automatic creation of sub-mapping methods
+
+MapStruct 1.2 will automatically create methods for nested mappings if possible.
+E.g. consider the following entities and corresponding DTOs:
+
+{{< prettify xml >}}
+public class Order {
+    public Customer customer;
+}
+
+public class Customer {
+    public String custName;
+}
+{{< /prettify >}}
+
+{{< prettify xml >}}
+public class OrderDto {
+    public CustomerDto customer;
+}
+
+public class CustomerDto {
+    public String name;
+}
+{{< /prettify >}}
+
+In MapStruct 1.1 you were required to provide two dedicated mapping methods, one for orders and one for customers:
+
+{{< prettify xml >}}
+@Mapper
+public class OrderMapper {
+
+    OrderDto orderToDto(Order order);
+
+    @Mapping(target = "name", source = "custName")
+    CustomerDto customerToDto(Customer customer);
+}
+{{< /prettify >}}
+
+If you have object graphs on source and target side that are structurally equal (i.e. for each object on the source side there is a corresponding object on the target side) but only differ in property names or (non-bean) property types,
+having to define mapping methods for all the source-target object pairs can be quite a bit of work.
+
+MapStruct 1.2 can therefore automatically create nested mapping methods:
+
+{{< prettify xml >}}
+@Mapper
+public class OrderMapper {
+
+    @Mapping(target = "customer.name", source = "customer.custName")
+    OrderDto orderToDto(Order order);
+}
+{{< /prettify >}}
+
+When detecting that there is no explicit method for mapping `Customer` to `CustomerDto`,
+MapStruct will generate such method, applying the `@Mapping` configuration given on the `orderToDto()` method.
+
+This feature should be used with care, but it can save you quite some work if you are mapping large graphs.
```

Comment:
We should explain why it should be used with care -- i.e. because it's new and we'd like the feedback from real applications -- basically what's in the second sentence.